erpnext/accounts/doctype/pos_closing_entry/pos_closing_entry.js

```diff
@@ -128,13 +128,13 @@ function refresh_payments(d, frm, remove) {
 	d.payments.forEach(p => {
 		const payment = frm.doc.payment_reconciliation.find(pay => pay.mode_of_payment === p.mode_of_payment);
 		if (payment) {
-			if (!remove) payment.expected_amount += flt(p.amount);
+			if (!remove) payment.expected_amount += flt(p.amount-d.change_amount);
```

Comment:
This won't work either. Since change amount will be subtracted from all mode of payments. 
If change is given via cash then only cash mode of payment should be subtracted by change amount. 

The problem is valid. However to solve it correctly we might have to check which mode of payment has same account as change amount's account. Then subtract the change amount from that one matching mode of payment.

Reply:
Well, I thank that's a rare case. Usually multiple mode of payments are made when one of them is limited / not sufficient. But I'm aware that my code is not a correct solution, as I'm not good at coding but I need the PoS system to work at my store. Could you please make a necessary changes at next update? Thankyou very much for your hard work in refactoring the PoS.